### PR TITLE
library grouped pagination (connects #390)

### PR DESCRIPTION
**Pull request by ABee-Tech**
### Description
Library page catagorization by resourse type

### Links and Curls
https://localhost:3000/library

### CODE REVIEW

**Comment by lmmrssa** [comment]
image

File: api/src/controllers/resourceController.js [hunk 1]
```diff
@@ -34,6 +34,39 @@ const getResources = (req, res) => {
     .catch((err) => res.json({ err }).status(400))
 }
 
+// @desc    Fetch all resources
+// @route   GET /api/resources/:category?pageNumber=${pageNumber}
+// @access  Public
+const getResourcesByCategory = (req, res) => {
```
Comment (lmmrssa) [inline api/src/controllers/resourceController.js:40]: this function is not needed only passing additional condition with req.category should work with getResources function

File: api/src/routes/resourceRouter.js [hunk 2]
```diff
@@ -7,9 +7,10 @@ const path = require('path')
 const protect = require('../middleware/authMiddleware')
 const { uploadArray, upload, resizeImage } = require('../helpers/filehelpers')
 
-const { getResources, addResource, getResourcesById, deleteResources, updateResources, searchResourcesTitle } = require('../controllers/resourceController.js')
+const { getResources, getResourcesByCategory, addResource, getResourcesById, deleteResources, updateResources, searchResourcesTitle } = require('../controllers/resourceController.js')
 
 router.route('/').get(getResources)
+router.route('/:category').get(getResourcesByCategory)
```
Comment (lmmrssa) [inline api/src/routes/resourceRouter.js:13]: this route is not needed

**Review by lmmrssa — CHANGES_REQUESTED** [review]

File: api/src/controllers/resourceController.js (hunk 1 above)
Comment (ABee-Tech) [reply api/src/controllers/resourceController.js]: Done

File: api/src/routes/resourceRouter.js (hunk 2 above)
Comment (ABee-Tech) [reply api/src/routes/resourceRouter.js]: Route removed

**Review by dogi — APPROVED** [review]

**Review by lmmrssa — APPROVED** [review]

**Pull request merged**